### PR TITLE
build: remove libfindmacros library

### PR DESCRIPTION
**Pull request by dundargoc**
Large parts the library weren't being used, and the parts that was were
was overly abstracted for our use case. Additionally, part of its use
case was to abstract pkgconfig boilerplate, which is no longer needed as
pkgconfig has been removed in favor of relying on cmake alone in
09118052cee5aef978d6075db5287c1b6c27381a.

### CODE REVIEW

File: cmake/LibFindMacros.cmake [hunk 1]
```diff
@@ -3,167 +3,36 @@
 # Maintained at https://github.com/Tronic/cmake-modules
```
Comment (justinmk) [inline cmake/LibFindMacros.cmake:3]: Does cmake 3.x include most of this now, or is it more like we don't use all/most of it?

Comment (dundargoc) [reply cmake/LibFindMacros.cmake:3]: I don't think we need it.

Comment (dundargoc) [reply cmake/LibFindMacros.cmake]: Elaborated on the commit message/pr description

**Comment by dundargoc** [comment]
@justinmk

**Review by justinmk — APPROVED** [review]
Beautiful! Less is more.

**Pull request merged**